src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -103,9 +111,23 @@ void GcsActorManager::RegisterActor(
 
   auto actor = std::make_shared<GcsActor>(request);
   RAY_CHECK(registered_actors_.emplace(actor->GetActorID(), actor).second);
+  if (actor->IsDetached()) {
+    RAY_CHECK(named_actors_.emplace(actor->GetName(), actor->GetActorID()).second);
```

Comment:
Hmm so do we want all detached actors to have names? I think it makes sense but just want to check whether this was true for the API before too.

Reply:
This was already the API, it's enforced in `actor.py`